### ISSUE TITLE
missing wrap_parameters: false on rails 7.2.

### ISSUE THREAD

**Issue by arikarim** [rails/rails#53047]
### Steps to reproduce
 missing `wrap_parameters: false` on rails 7.2.
 
 after upgrading to rails 7.2, i have encountered a weird issue of parameter missing.
 after investigating i found out that rails was wrapping the params inside the controller name, for example:
 sending:
 ```
{
    "purchaseRequisition": {
				"data that will be missing because of wrap parameters": "not available"
    }
}
``` 
was resulting:
```
#<ActionController::Parameters {"purchaseRequisition"=>{"data that will be missing because of wrap parameters"=>"not available"}, "controller"=>"v1/purchase_requisitions", "action"=>"create", "purchase_requisition"=>{}} permitted: false>
```

as you can see rails already have wrapped the params and have a key `purchase_requisition` for it.

so if by any chance you send data like above and call deep_transform_keys on it the result will be:
```
#<ActionController::Parameters {"purchase_requisition"=>{}, "controller"=>"v1/purchase_requisitions", "action"=>"create"} permitted: false>
```

**Comment by fatkodima** [comment]
Can you provide a reproduction test script using https://github.com/rails/rails/blob/main/guides/bug_report_templates/action_controller.rb template?

**Comment by zzak** [comment]
Maybe fixed by #52763, can you check it?
